### PR TITLE
Update xtensa LLVM version into 15.x

### PR DESCRIPTION
**Pull request by dongsheng28849455**
Fix the issue that xtensa .aot size cannot be over 256K

### CODE REVIEW

File: core/iwasm/aot/arch/aot_reloc_xtensa.c [hunk 1]
```diff
@@ -257,9 +257,11 @@ apply_relocation(AOTModule *module, uint8 *target_section_addr,
 
             /* check relative offset boundary */
             if (relative_offset < -256 * BH_KB || relative_offset > -4) {
-                set_error_buf(error_buf, error_buf_size,
-                              "AOT module load failed: "
-                              "target address out of range.");
+                set_error_buf(
+                    error_buf, error_buf_size,
+                    "AOT module load failed: "
+                    "target address out of range. \r\n"
```
Comment (wenyongh) [inline core/iwasm/aot/arch/aot_reloc_xtensa.c:263]: Had better remove the trailing spaces: "target address out of range.\r\n"

File: core/iwasm/aot/arch/aot_reloc_xtensa.c [hunk 2]
```diff
@@ -257,9 +257,11 @@ apply_relocation(AOTModule *module, uint8 *target_section_addr,
...
+                    "use --size-level=0 option to generate .literal island");
```
Comment (wenyongh) [inline core/iwasm/aot/arch/aot_reloc_xtensa.c:264]: How about changing `use` to `Try using`?

File: core/iwasm/aot/arch/aot_reloc_xtensa.c [hunk 3]
```diff
@@ -257,9 +257,11 @@ apply_relocation(AOTModule *module, uint8 *target_section_addr,
...
             /* check relative offset boundary */
             if (relative_offset < -256 * BH_KB || relative_offset > -4) {
-                set_error_buf(error_buf, error_buf_size,
-                              "AOT module load failed: "
-                              "target address out of range.");
+                set_error_buf(
+                    error_buf, error_buf_size,
+                    "AOT module load failed: "
+                    "target address out of range.\n"
+                    "Try using `wamrc --size-level=0` to generate .literal island");
```
Comment (wenyongh) [inline core/iwasm/aot/arch/aot_reloc_xtensa.c:264]: Line is too long, code guideline CI ran failed.

Comment (dongsheng28849455) [reply core/iwasm/aot/arch/aot_reloc_xtensa.c]: sorry, update again

**Pull request merged**